applications/asset_tracker_v2/src/modules/modem_module.c

```diff
@@ -559,135 +559,36 @@ static int static_modem_data_get(void)
-			[sizeof(event->data.modem_dynamic.apn) - 1] = '\0';
-
-		prev.apn[sizeof(prev.apn) - 1] = '\0';
-
-		event->data.modem_dynamic.apn_fresh = true;
-		params_added = true;
-	}
-
-	if ((strcmp(prev.ip_address, param->network.ip_address.value_string) != 0) || include) {
-		strncpy(event->data.modem_dynamic.ip_address,
-			modem_param.network.ip_address.value_string,
-			sizeof(event->data.modem_dynamic.ip_address) - 1);
-
-		strncpy(prev.ip_address,
-			param->network.ip_address.value_string,
-			sizeof(prev.ip_address) - 1);
-
-		event->data.modem_dynamic.ip_address
-			[sizeof(event->data.modem_dynamic.ip_address) - 1] = '\0';
-
-		prev.ip_address[sizeof(prev.ip_address) - 1] = '\0';
-
-		event->data.modem_dynamic.ip_address_fresh = true;
-		params_added = true;
-	}
-
-	if ((prev.cell_id != param->network.cellid_dec) || include) {
-		event->data.modem_dynamic.cell_id = param->network.cellid_dec;
-		prev.cell_id = param->network.cellid_dec;
-
-		event->data.modem_dynamic.cell_id_fresh = true;
-		params_added = true;
-	}
-
-	if ((strcmp(prev.mccmnc, param->network.current_operator.value_string) != 0) || include) {
-		strncpy(event->data.modem_dynamic.mccmnc,
-			modem_param.network.current_operator.value_string,
-			sizeof(event->data.modem_dynamic.mccmnc));
-
-		strncpy(prev.mccmnc, param->network.current_operator.value_string,
-			sizeof(prev.mccmnc));
-
-		event->data.modem_dynamic.mccmnc
-			[sizeof(event->data.modem_dynamic.mccmnc) - 1] = '\0';
-
-		prev.mccmnc[sizeof(prev.mccmnc) - 1] = '\0';
-
-		/* Provide MNC and MCC as separate values. */
-		event->data.modem_dynamic.mcc = modem_param.network.mcc.value;
-		event->data.modem_dynamic.mnc = modem_param.network.mnc.value;
-
-		event->data.modem_dynamic.mccmnc_fresh = true;
-		params_added = true;
-	}
-
-	if ((prev.area_code != modem_param.network.area_code.value) || include) {
-		event->data.modem_dynamic.area_code = param->network.area_code.value;
-		prev.area_code = param->network.area_code.value;
-
-		event->data.modem_dynamic.area_code_fresh = true;
-		params_added = true;
-	}
-
-	if (params_added) {
-		event->type = MODEM_EVT_MODEM_DYNAMIC_DATA_READY;
-		event->data.modem_dynamic.timestamp = k_uptime_get();
-	} else {
-		LOG_DBG("No dynamic modem parameters have changed from the last sample request.");
-		event->type = MODEM_EVT_MODEM_DYNAMIC_DATA_NOT_READY;
-	}
+	event->data.modem_dynamic.rsrp = rsrp_value_latest;
+	event->data.modem_dynamic.band = (uint8_t)param->network.current_band.value;
+	event->data.modem_dynamic.nw_mode = nw_mode_latest;
+	event->data.modem_dynamic.mcc = modem_param.network.mcc.value;
+	event->data.modem_dynamic.mnc = modem_param.network.mnc.value;
+	event->data.modem_dynamic.area_code = param->network.area_code.value;
+	event->type = MODEM_EVT_MODEM_DYNAMIC_DATA_READY;
+	event->data.modem_dynamic.timestamp = k_uptime_get();
+	event->data.modem_dynamic.cell_id = (uint32_t)param->network.cellid_dec;
+
+	strncpy(event->data.modem_dynamic.apn,
```

Comment:
could/should we do the copying and NULL-terminating in one go using snprintf?

Reply:
I assume it's done this way to guarantee null termination, at the cost of possibly silently cutting strings short